docs/proposals/20220209-machinepool-machines.md

```diff
@@ -0,0 +1,196 @@
+|----|-------------|-----------------|
+| R1 | The MachinePool controller MUST create Machines representing the provider-specific resources in the MachinePool and enable cascading delete of infrastructure<br>machine resources upon delete of CAPI Machine. | U1 |
+| R2 | The machine health check controller MUST be able to select machines by label which belong to a MachinePool and remediate. | U2 |
+| R3 | The MachinePool API MUST provide an optional deployment strategy using the same type as MachineDeployment.Spec.Strategy. | U3 |
+| R4 | The Machine Controller MUST handle node drain for Machine Pool Machines with the same behavior as MachineDeployment Machines. | U1, U3 |
+
+### Implementation Details/Notes/Constraints
+
+- As an alternative, the `InfrastructureSelector` field could be attached to the CAPI MachinePool resource. Then the CAPI MachinePool controller could check it directly and it would be more clearly documented in code.
+
+- Since this is an optional feature, some providers will implement MachinePool Machines and others will not. While having MachinePool Machines represented improves the consistency of the UX when compared to MachineDeployments, they also create a split in the UX between different MachinePool providers: some will show Machines and others won't (the current behavior).
+
+  One alternative to that would be to create "hollow" MachinePool Machines, simply to have a representation of how many instances or replicas are in the MachinePool. These Machines would either be prevented from deletion, or would have no effect on actual provider resources when deleted. This doesn't seem like an obvious improvement over the previous option.
+
+  Since MachinePools are experimental and the several implementations are well-known, we expect that providers will implement MachinePool Machines fairly soon, and that the only implementations that cannot support it are AKSManagedMachinePool and AWSManagedMachinePool. The variation in UX is unfortunate but at least limited to these special types of MachinePool where it wraps a managed service. In those cases the behavior is likely to be understood by the user if they are familiar with the provider's service.
+
+### Risks and Mitigations
+
+- A previous version of this proposal suggested using a pre-populated `infrastructureRefList` instead of the `InfrastructureSelector` and related fields. Effectively, this means the query "what provider resources exist for this MachinePool's instances?" would be pre-populated, rather than run as needed. This approach would not scale well to MachinePools with hundreds or thousands of instances, since that list becomes part of the representation of that MachinePool, possibly causing storage or network transfer issues.
+
+  It was pointed out that the existing `providerIDList` suffers the same scalability issue as an `infrastructureRefList` would. It would be nice to refactor it away, but doing so is not considered a goal of this proposal effort.
```

Comment:
should we open an issue for this if there isn't already one at there, and if there is, link it here?

Reply:
See #6387.